test/built-ins/Temporal/PlainDate/prototype/toString/order-of-operations.js

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindate.prototype.tostring
+description: Properties on an object passed to toString() are accessed in the correct order
+includes: [compareArray.js, temporalHelpers.js]
+features: [Temporal]
+---*/
+
+const expected = [
+  "get options.calendarName",
+  "get options.calendarName.toString",
+  "call options.calendarName.toString",
+  "get this.calendar[Symbol.toPrimitive]",
+  "get this.calendar.toString",
+  "call this.calendar.toString",
+];
+const actual = [];
+
+const calendar = TemporalHelpers.calendarObserver(actual, "this.calendar");
+const instance = new Temporal.PlainDate(2000, 5, 2, calendar);
+// clear observable operations that occurred during the constructor call
+actual.splice(0, actual.length);
```

Comment:
FYI, the second parameter is optional and will have this behavior if omitted - with less room for error. Could be nice to prefer actual.splice(0), or perhaps this is worth moving out of TemporalHelpers into a more broad observer helper?

Reply:
Sure, I'll add another commit to clean up existing uses of this.

As for an observer helper, I'm hoping that tests like these will help build a strong case for such a helper in the future via the RFC process.